printer.go

```diff
@@ -136,6 +173,120 @@ type DriverInfo struct {
 	Attributes  uint32
 }
 
+// PrintJobInfo stores information about a print job.
+type PrintJobInfo struct {
+	JobID           uint32
+	UserMachineName string
+	UserName        string
+	DocumentName    string
+	DataType        string
+	Status          string
+	StatusCode      uint32
+	Priority        uint32
+	Position        uint32
+	TotalPages      uint32
+	PagesPrinted    uint32
+	Submitted       time.Time
+}
+
+// Jobs returns information about all print jobs on this printer
+func (p *Printer) Jobs() ([]PrintJobInfo, error) {
+	var bytesNeeded, jobsReturned uint32
+	buf := make([]byte, 1)
+	for {
+		err := EnumJobs(p.h, 0, 255, 1, &buf[0], uint32(len(buf)), &bytesNeeded, &jobsReturned)
+		if err == nil {
+			break
+		}
+		if err != syscall.ERROR_INSUFFICIENT_BUFFER {
+			return nil, err
+		}
+		if bytesNeeded <= uint32(len(buf)) {
+			return nil, err
+		}
+		buf = make([]byte, bytesNeeded)
+	}
+	if jobsReturned <= 0 {
+		return nil, nil
+	}
+	pj := make([]PrintJobInfo, 0, jobsReturned)
+	ji := (*[2048]JOB_INFO_1)(unsafe.Pointer(&buf[0]))[:jobsReturned]
+	for idx, j := range ji {
+		pj = append(pj, PrintJobInfo{
+			JobID:        j.JobID,
+			StatusCode:   j.StatusCode,
+			Priority:     j.Priority,
+			Position:     j.Position,
+			TotalPages:   j.TotalPages,
+			PagesPrinted: j.PagesPrinted,
+		})
+		// get strings, checking for nil
+		if j.MachineName != nil {
+			pj[idx].UserMachineName = syscall.UTF16ToString((*[2048]uint16)(unsafe.Pointer(j.MachineName))[:])
+		}
+		if j.UserName != nil {
+			pj[idx].UserName = syscall.UTF16ToString((*[2048]uint16)(unsafe.Pointer(j.UserName))[:])
+		}
+		if j.Document != nil {
+			pj[idx].DocumentName = syscall.UTF16ToString((*[2048]uint16)(unsafe.Pointer(j.Document))[:])
+		}
+		if j.DataType != nil {
+			pj[idx].DataType = syscall.UTF16ToString((*[2048]uint16)(unsafe.Pointer(j.DataType))[:])
+		}
+		if j.Status != nil {
+			pj[idx].Status = syscall.UTF16ToString((*[2048]uint16)(unsafe.Pointer(j.Status))[:])
+		}
+		if strings.TrimSpace(pj[idx].Status) == "" {
+			switch j.StatusCode {
```

Comment:
This code does not look right. I run your new test here, and see this:
```
    printer_test.go:100: {
           "JobID": 3,
           "UserMachineName": "\\\\UUUU",
           "UserName": "Alex",
           "DocumentName": "my document",
           "DataType": "RAW",
           "Status": "Unknown",
           "StatusCode": 8210,
           "Priority": 1,
           "Position": 1,
           "TotalPages": 1,
           "PagesPrinted": 0,
           "Submitted": "2018-10-06T13:34:01.000021392Z"
        }
    printer_test.go:100: {
           "JobID": 4,
           "UserMachineName": "\\\\UUUU",
           "UserName": "Alex",
           "DocumentName": "my document",
           "DataType": "RAW",
           "Status": "Unknown",
           "StatusCode": 0,
           "Priority": 1,
           "Position": 2,
           "TotalPages": 1,
           "PagesPrinted": 0,
           "Submitted": "2018-10-06T13:34:21.000020568Z"
        }
```
and I wondered why Status says 'Unknown'? I think you are misinterpreting the JOB_INFO_1 doco, which says:
```
The job status. The value of this member can be zero or a combination of one or more of the following values. A value of zero indicates that the print queue was paused after the document finished spooling.
```
- you are not handling Status field being 0;
- you are not treating Status value as "combination of one or more of the following values.".